engine/evd.ml

```diff
@@ -466,6 +464,19 @@ type side_effects = {
   seff_roles : side_effect_role Cmap.t;
 }
 
+type future_goals = {
```

Comment:
Is it possible to make this type private? It doesn't seem to be created outside of this API.